shepherd_bms/lib/LTC/LTC68041.cpp

```diff
@@ -306,13 +306,20 @@ uint8_t LTC6804_rdcv(uint8_t reg, // Controls which cell voltage register is rea
           //must increment by two for each parsed cell code
         }
         //a.iii
-        received_pec = (cell_data[data_counter] << 8) + cell_data[data_counter+1]; //The received PEC for the current_ic is transmitted as the 7th and 8th
         //after the 6 cell voltage data bytes
         data_pec = pec15_calc(BYT_IN_REG, &cell_data[current_ic * NUM_RX_BYT]);
-        if (received_pec != data_pec)
-        {
-          pec_error = -1;                             //The pec_error variable is simply set negative if any PEC errors
-          //are detected in the serial data
+        uint8_t RETRY_COUNT = 0;
+        while (RETRY_COUNT < LTC_MAX_RETRIES) {
```

Comment:
I'm a bit sketched out by a while loop here. I know I said to use it but it might not be the best option, especially since getting data takes a few milliseconds. A while loop will block all code for a lil bit, and it might be cleaner to just return an error code if we can't get data. Or we can lower the max retries to 2 or some lower number

Reply:
In that situation, couldn't we keep the previous code returning -1? No change required